splinterd/src/main.rs

```diff
@@ -302,11 +318,20 @@ fn main() {
         .map(String::from)
         .or_else(|| config.registry_file());
 
+    // Allow unused mut for experimental features
+    #[allow(unused_mut)]
+    let mut feature_fields = "".to_string();
```

Comment:
This is debug output, so it should only be done within the `debug!` macro's call.

You should replace `feature_fields` in the debug log statement with something like:
```rust
        ...
        heartbeat_interval,
        if cfg!(feature = "database") {
            format!("{}, db_url: {}", feature_fields, db_url)
        } else {
            String::new()
        }
    );
```

Reply:
updated